sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala

```diff
@@ -217,13 +271,91 @@ trait Interpreter extends ScorexLogging {
       val prop = propositionFromErgoTree(tree, context1)
 
       implicit val vs = context1.validationSettings
-      val (propTree, context2) = trySoftForkable[(BoolValue, CTX)](whenSoftFork = (TrueLeaf, context1)) {
+      val (propTree, context2) = trySoftForkable[(SigmaPropValue, CTX)](whenSoftFork = (TrueLeaf, context1)) {
+        applyDeserializeContext(context1, prop)
+      }
+
+      // here we assume that when `propTree` is TrueProp then `reduceToCrypto` always succeeds
+      // and the rest of the verification is also trivial
+      val (cProp, cost) = reduceToCrypto(context2, env, propTree, (context2.initCost - context.initCost).toInt).getOrThrow
```

Comment:
Looks like consensus-related issue, isn't it ?

Reply:
Which issue you mean?

Reply:
change in the cost calculated. then old Sigma-3.x powered node and updated one will diverge in block cost estimation, which could result in disagreement on block validation and network split as result.

Reply:
no, all the historical blocks will be valid with respect to JIT. This will be ensured during [forking process](https://github.com/ergoplatform/eips/blob/5f9d14dabe38828f083f72b88bf828e2e78d7dd6/eip-0008.md#selection-of-hard-forking-block).

Reply:
@aslesarenko I don't see any conditional logic around "context2.initCost - context.initCost"

Reply:
Rolled back all the changes in reduceToCrypto. New version moved to reduceToCryptoJITC and it is not used by default.